### PR TITLE
Fixes for #1089

### PR DESCRIPTION
**Pull request by ShikharJ**
Renamed UndefinedError to DomainError and made the necessary changes as directed.

### CODE REVIEW

**Comment by ShikharJ** [comment]
The build failed on a single test #4061.12 on Travis CI. If @certik or @isuruf could help me with this, it would be a great favour.

File: symengine/real_mpfr.cpp [hunk 1]
```diff
@@ -86,7 +86,7 @@ RCP<const Number> RealMPFR::addreal(const Complex &other) const
     mpc_add_fr(t.get_mpc_t(), t.get_mpc_t(), this->i.get_mpfr_t(), MPFR_RNDN);
     return complex_mpc(std::move(t));
 #else
-    throw SymEngineException("Result is complex. Recompile with MPC support.");
+    throw DomainError("Result is complex. Recompile with MPC support.");
```
Comment (isuruf) [inline symengine/real_mpfr.cpp]: On second thought, this should just be `SymEngineException`

Comment (ShikharJ) [inline symengine/real_mpfr.cpp]: Umm, so should i keep both or revert to original?

**Comment by isuruf** [comment]
For the travis-ci failure, see, https://travis-ci.org/symengine/symengine/jobs/170506482#L663-L680

File: symengine/real_mpfr.cpp (hunk 1 above)
Comment (isuruf) [inline symengine/real_mpfr.cpp]: Revert to original

Comment (ShikharJ) [inline symengine/real_mpfr.cpp]: Thanks Sir! I'll keep you posted if I run into another failure.

**Comment by ShikharJ** [comment]
I am altogether facing a lot of issues. I guess, I'll start a fresh pull request soon.

**Pull request closed without merging**